core/modules/mpls_push.cc

```diff
@@ -0,0 +1,52 @@
+#include "mpls_push.h"
+
+#include "../utils/endian.h"
+#include "../utils/ether.h"
+#include "../utils/mpls.h"
+
+using bess::utils::Ethernet;
+using bess::utils::be16_t;
+using bess::utils::Mpls;
+
+const Commands MPLSPush::cmds = {{"set", "MplsPushArg",
+                                  MODULE_CMD_FUNC(&MPLSPush::CommandSet),
+                                  Command::THREAD_UNSAFE}};
+
+MPLSPush::MPLSPush() : label_(0), ttl_(64), tc_(0), is_bottom_of_stack_(true) {}
+
+
+CommandResponse MPLSPush::Init(const bess::pb::MplsPushArg &arg) {
+  return CommandSet(arg);
+}
+
+void MPLSPush::ProcessBatch(bess::PacketBatch *batch) {
+  int cnt = batch->cnt();
+  for (int i = 0; i < cnt; i++) {
+    bess::Packet *pkt = batch->pkts()[i];
+    Ethernet *eth = pkt->head_data<Ethernet *>();
+
+    Ethernet::Address src_addr = eth->src_addr;
+    Ethernet::Address dst_addr = eth->dst_addr;
+
+    pkt->prepend(4);
```

Comment:
All code south of here should be guarded by a check that this call to `Packet::prepend()` didn't fail.